### ISSUE TITLE
False positive scss/operator-no-unspaced  Expected single space after "-"

### ISSUE THREAD

**Issue by jkevingutierrez** [stylelint-scss/stylelint-scss#561]
Input:

```
@mixin test {
  @for $i from 0 through 10 {
    &:nth-child(#{$i + 1}) {
      top: #{$i * -10}px;
      transform: scale(#{(10 - $i) * 0.1}) !important;
    }
  }
}
```
Require:

```
@mixin test {
  @for $i from 0 through 10 {
    &:nth-child(#{$i + 1}) {
      top: #{$i * - 10}px;
      transform: scale(#{(10 - $i) * 0.1}) !important;
    }
  }
}
```
Expected (no errors):

```
@mixin test {
  @for $i from 0 through 10 {
    &:nth-child(#{$i + 1}) {
      top: #{$i * -10}px;
      transform: scale(#{(10 - $i) * 0.1}) !important;
    }
  }
}
```

**Comment by stefanfisk** [comment]
I'm getting a bunch of `Expected single space before "+"  scss/operator-no-unspaced` and `Expected single space after "+"   scss/operator-no-unspaced` for 

```
background: url(data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA0Ljk1IDEwIj48ZGVmcz48c3R5bGU+LmNscy0xe2ZpbGw6I2ZmZjt9LmNscy0ye2ZpbGw6IzQ0NDt9PC9zdHlsZT48L2RlZnM+PHRpdGxlPmFycm93czwvdGl0bGU+PHJlY3QgY2xhc3M9ImNscy0xIiB3aWR0aD0iNC45NSIgaGVpZ2h0PSIxMCIvPjxwb2x5Z29uIGNsYXNzPSJjbHMtMiIgcG9pbnRzPSIxLjQxIDQuNjcgMi40OCAzLjE4IDMuNTQgNC42NyAxLjQxIDQuNjciLz48cG9seWdvbiBjbGFzcz0iY2xzLTIiIHBvaW50cz0iMy41NCA1LjMzIDIuNDggNi44MiAxLjQxIDUuMzMgMy41NCA1LjMzIi8+PC9zdmc+)
```

**Comment by kristerkari** [comment]
> Expected (no errors):
> 
> ```
> @mixin test {
>   @for $i from 0 through 10 {
>     &:nth-child(#{$i + 1}) {
>       top: #{$i * -10}px;
>       transform: scale(#{(10 - $i) * 0.1}) !important;
>     }
>   }
> }
> ```

@jkevingutierrez I opened a pull request to fix this one: https://github.com/stylelint-scss/stylelint-scss/pull/575

**Comment by kristerkari** [comment]
> I'm getting a bunch of `Expected single space before "+" scss/operator-no-unspaced` and `Expected single space after "+" scss/operator-no-unspaced` for
> 
> ```
> background: url(data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA0Ljk1IDEwIj48ZGVmcz48c3R5bGU+LmNscy0xe2ZpbGw6I2ZmZjt9LmNscy0ye2ZpbGw6IzQ0NDt9PC9zdHlsZT48L2RlZnM+PHRpdGxlPmFycm93czwvdGl0bGU+PHJlY3QgY2xhc3M9ImNscy0xIiB3aWR0aD0iNC45NSIgaGVpZ2h0PSIxMCIvPjxwb2x5Z29uIGNsYXNzPSJjbHMtMiIgcG9pbnRzPSIxLjQxIDQuNjcgMi40OCAzLjE4IDMuNTQgNC42NyAxLjQxIDQuNjciLz48cG9seWdvbiBjbGFzcz0iY2xzLTIiIHBvaW50cz0iMy41NCA1LjMzIDIuNDggNi44MiAxLjQxIDUuMzMgMy41NCA1LjMzIi8+PC9zdmc+)
> ```

@stefanfisk I opened a pull request with a fix for this one: https://github.com/stylelint-scss/stylelint-scss/pull/574

**Comment by kristerkari** [comment]
Both cases are fixed in the latest 4.0.1 version
